### ISSUE TITLE
Error Creating Cluster in OpenShift: fsGroup: Invalid value: []int64{26}: 26 is not an allowed group

### ISSUE THREAD

**Issue by dmhumph** [CrunchyData/postgres-operator#517]
**Describe the bug**
I am running OpenShift 3.11 and using pgo 3.4 from the quickstart.sh script.  Everything deploys fine initially until I try to deploy a cluster.  The error back from OpenShift is: Error creating: pods "mycluster-59dbc4cdb-" is forbidden: unable to validate against any security context constraint: [fsGroup: Invalid value: []int64{26}: 26 is not an allowed group]

**To Reproduce**
1. run quickstart.sh on linux host
2. specify gp2 as storage class to use in quickstart.sh
2. run pgo create cluster mycluster
3. oc describe rs/mycluster-59dbc4cdb
4. Rececive the following error:
Events:
  Type     Reason        Age               From                   Message
  ----     ------        ----              ----                   -------
  Warning  FailedCreate  2m (x17 over 7m)  replicaset-controller  Error creating: pods "mycluster-59dbc4cdb-" is forbidden: unable to validate against any security context constraint: [fsGroup: Invalid value: []int64{26}: 26 is not an allowed group]
5. Cluster is not created

**Expected behavior**
I expect that a simple cluster would be created.

**Please tell us about your environment:**

* Operating System:
RHEL 7.5
* Where is this running ( Local, Cloud Provider)
AWS
* Storage being used (NFS, Hostpath, Gluster, etc):
EBS
* Container Image Tag:
crunchydata/crunchy-postgres:centos7-10.6-2.2.0
* Platform (Docker, Kubernetes, OpenShift):
OpenShift
* Platform Version:
3.11

**Additional context**
I believe the quickstart script needs to have a step to add scc to allow permissions.

**Comment by jmccormick2001** [comment]
correct, the quickstart script assumes the cluster admin has allowed fsgroup 26 in the restricted scc, but we have been thinking of adding a scc specific to the operator that people can install with, we might go that way eventually.

**Comment by dmhumph** [comment]
Is that documented somewhere for OpenShift that a user with the cluster-admin role needs to allow fsgroup 26 in the restricted scc?  
Something like "In OpenShift a user with cluster-admin role needs to run the following command before running the quickstart.sh" 
And is this the proper command?
oc adm policy add-scc-to-group restricted 26
where restricted is the scc name and 26 is the group name.

**Comment by jmccormick2001** [comment]
fsGroup:
  type: RunAsAny

I change that as the cluster admin by running:
oc edit scc restricted

then I make that change, this is for development/testing, for a real production deployment
you would likely reduce the scope of that range, in the OCP docs they give an example
of specifying a set range of GID's.

People like you say typically however will create a dedicated scc for various apps
to work against to avoid having to touch the restricted scc, so that is something advanced
users might want to explore, or we might provide that scc as part of the setup/install
as well.. I like that idea but haven't got around to building the scc.

The docs or the quickstart script should mentioned the above reqt which it does not, it assumes
its been done already which might be a trip-up for most people.

**Comment by dmhumph** [comment]
Thank you.  I will see about creating a custom scc to work in this setup.  If I get something working that is more secure than RunAsAny I am happy to add to the documentation on the steps.

**Comment by Klaas-** [comment]
> 
> 
> fsGroup:
> type: RunAsAny
> 
> I change that as the cluster admin by running:
> oc edit scc restricted


This will stop openshift from upgrading. it will present this message:

`Changes to bootstrapped SCCs have been detected. Please review the changes by running "oc adm policy --config=/etc/origin/master/admin.kubeconfig reconcile-sccs --additive-only=true" After reviewing the changes please apply those changes by adding the '--confirm' flag. Do not modify the default SCCs. Customizing the default SCCs will cause this check to fail when upgrading. If you require non standard SCCs please refer to https://docs.okd.io/latest/admin_guide/manage_scc.html`

**Comment by mmgaggle** [comment]
@dmhumph - did ever manage to get a working scc?

**Comment by dmhumph** [comment]
@mmgaggle Yes, my process to get this working was to create a new scc in openshift.  It has been a little while since I last did this, but I need to try it out again and commit my documentation for OpenShift 3.11.  Although I have a feeling some of this will change in OCP 4.x

**Comment by mmgaggle** [comment]
@dmhumph I'd be happy to test out what you have for 3.11 on my 4.1 cluster

**Comment by mmgaggle** [comment]
Example SCC that seems to work for me.

```
allowHostDirVolumePlugin: false
allowHostIPC: false
allowHostNetwork: false
allowHostPID: false
allowHostPorts: false
allowPrivilegeEscalation: true
allowPrivilegedContainer: false
allowedCapabilities: null
apiVersion: security.openshift.io/v1
defaultAddCapabilities: null
fsGroup:
  type: RunAsAny
groups:
- system:authenticated
kind: SecurityContextConstraints
metadata:
  annotations:
    kubernetes.io/description: scc for postgres
  creationTimestamp: "2019-07-18T14:07:50Z"
  generation: 3
  name: pgo
priority: null
readOnlyRootFilesystem: false
requiredDropCapabilities:
- KILL
- MKNOD
- SETUID
- SETGID
runAsUser:
  type: MustRunAsRange
seLinuxContext:
  type: RunAsAny
supplementalGroups:
  type: RunAsAny
users: []
volumes:
- configMap
- downwardAPI
- emptyDir
- persistentVolumeClaim
- projected
- secret
```

**Comment by jmccormick2001** [comment]
thanks, I've added that scc to the examples directory here:

https://github.com/CrunchyData/postgres-operator/blob/develop/examples/pgo-scc.yaml

**Comment by dmhumph** [comment]
Sorry, this fell off my task list.... so the way I solved this was in a little more secure manner by specifying a fsgroup range as outlined in the following document:
https://docs.openshift.com/container-platform/3.11/install_config/persistent_storage/pod_security_context.html#fsgroup

I copied the restricted scc and then modified it to include fsgroup and ranges between 20 and 30.  

Unfortunately I have not tested lately but those are the steps I took.  And given time I will test in the next couple days and post the scc.
